rust/kernel/module_param.rs

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Types for module parameters.
+//!
+//! C header: [`include/linux/moduleparam.h`](../../../include/linux/moduleparam.h)
+
+use core::fmt::Write;
+
+/// Types that can be used for module parameters.
+/// Note that displaying the type in `sysfs` will fail if `to_string` returns
+/// more than `kernel::PAGE_SIZE` bytes (including an additional null terminator).
+pub trait ModuleParam: core::fmt::Display + core::marker::Sized {
+    /// Setting this to `true` allows the parameter to be passed without an
+    /// argument (e.g. just `module.param` instead of `module.param=foo`).
+    const NOARG_ALLOWED: bool;
+
+    /// `arg == None` indicates that the parameter was passed without an
+    /// argument. If `NOARG_ALLOWED` is set to `false` then `arg` is guaranteed
+    /// to always be `Some(_)`.
+    fn try_from_param_arg(arg: Option<&[u8]>) -> Option<Self>;
```

Comment:
Summary line missing, i.e. to answer what the implementation is supposed to fill here?